MapboxCoreNavigationTests/RouteControllerTests.swift

```diff
@@ -113,4 +134,69 @@ class RouteControllerTests: XCTestCase {
         
         XCTAssertEqual(navigation.location!.course, mbTestHeading, "Course should be using bearing")
     }
+
+    func testUserWithinTunnelEntranceRadius() {
+        let navigation = tunnelSetup.routeController
+        
+        // Intersection with a tunnel roadClass
+        let intersectionLocation = navigation.routeProgress.route.legs[0].steps[1].intersections![1].location
+        let tunnelEntranceLocation = CLLocation(latitude: intersectionLocation.latitude, longitude: intersectionLocation.longitude)
+        
+        navigation.locationManager(navigation.locationManager, didUpdateLocations: [tunnelEntranceLocation])
+
+        let currentLegProgress = (navigation.routeProgress.currentLegProgress)!
+        let currentStep = currentLegProgress.currentStep
+        let currentIntersection = currentStep.intersections![currentLegProgress.stepIndex]
+        
+        var currentLocation = location(at: navigation.location!.coordinate)
+        var distanceToTunnelEntrance = currentIntersection.location.distance(to: currentLocation.coordinate)
+        
+        var userIsAtTunnelEntranceRadius = navigation.userWithinTunnelEntranceRadius(at: currentLocation, intersection: currentIntersection, distance: distanceToTunnelEntrance)
```

Comment:
Looking at `userWithinTunnelEntranceRadius`, why is it necessary to pass in `distance`? If we're passing in the user's currentLocation and an intersection, can't we already calculate the distance inside this function with these arguments?

Reply:
It's a preference at this point. You can pass it in as a parameter or let the function do calculation. I rather do the former -- making the function lightweight.

Reply:
But there is no reason to pass it in. I think fewer arguments would making this function easier.

Reply:
@bsudekum I think you misunderstood me. When I say, light-weight - it's in regards to the function scope complexity (not parameters). Besides, I like how we just provide all the values and allow the function to just compare and examine values and send a `state`. Just my 2¢

Reply:
I think I'm with Vince here - IMO having this clearly outlined in the method signature improves the clarity of intent and potentially reduces implicit coupling.

Reply:
Per pairing, it has to be this way since calculating distance inside the function would either calculate absolute distance or require the addition of a coords argument.